apps/jobs/views.py

```diff
@@ -624,6 +625,152 @@ def leaderboard(request, challenge_phase_split_id):
     return paginator.get_paginated_response(response_data)
 
 
+@api_view(["GET"])
+@throttle_classes([AnonRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_all_entries_on_public_leaderboard(request, challenge_phase_split_pk):
+    """
+    Returns public/private leaderboard entries to corresponding challenge phase split for a challenge host
+
+    - Arguments:
+        ``challenge_phase_split_pk``: Primary key for the challenge phase split for which leaderboard is to be fetched
+
+    - Returns:
+        All Leaderboard entry objects in a list
+    """
+    # check if the challenge exists or not
+    challenge_phase_split = get_challenge_phase_split_model(challenge_phase_split_pk)
+
+    # Get the leaderboard associated with the Challenge Phase Split
+    leaderboard = challenge_phase_split.leaderboard
+
+    # Get the default order by key to rank the entries on the leaderboard
+    try:
+        default_order_by = leaderboard.schema["default_order_by"]
+    except KeyError:
+        response_data = {
+            "error": "Sorry, Default filtering key not found in leaderboard schema!"
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    # Exclude the submissions done by members of the host team
+    # while populating leaderboard
+    challenge_obj = challenge_phase_split.challenge_phase.challenge
+
+    # Allow access only to challengeHost
```

Comment:
`# Allow access only to challengeHost` ->`# Allow access only to challenge host`